Bio/SeqUtils/IsoelectricPoint.py

```diff
@@ -114,69 +111,51 @@ def _update_pKs_tables(self):
             neg_pKs["Cterm"] = pKcterminal[cterm]
         return pos_pKs, neg_pKs
 
-    # This function calculates the total charge of the protein at a given pH.
-    def _chargeR(self, pH):
+    def charge_at_pH(self, pH):
+        """Calculate the charge of a protein at given pH."""
+        # derivation:
+        #   Henderson Hasselbalch equation: pH = pKa + log([A-]/[HA])
+        #   Rearranging: [HA]/[A-] = 10 ** (pKa - pH)
+        #   partial_charge =
+        #       [A-]/[A]total = [A-]/([A-] + [HA]) = 1 / { ([A-] + [HA])/[A-] } =
+        #       1 / (1 + [HA]/[A-]) = 1 / (1 + 10 ** (pKa - pH)) for acidic residues;
+        #                             1 / (1 + 10 ** (pH - pKa)) for basic residues
         positive_charge = 0.0
         for aa, pK in self.pos_pKs.items():
-            CR = 10 ** (pK - pH)
-            partial_charge = CR / (CR + 1.0)
+            partial_charge = 1.0 / (10 ** (pH - pK) + 1.0)
             positive_charge += self.charged_aas_content[aa] * partial_charge
 
         negative_charge = 0.0
         for aa, pK in self.neg_pKs.items():
-            CR = 10 ** (pH - pK)
-            partial_charge = CR / (CR + 1.0)
+            partial_charge = 1.0 / (10 ** (pK - pH) + 1.0)
             negative_charge += self.charged_aas_content[aa] * partial_charge
 
         return positive_charge - negative_charge
 
-    def charge_at_pH(self, pH):
-        """Calculate the charge of a protein at given pH."""
-        return self._chargeR(pH)
-
     # This is the action function, it tries different pH until the charge of
     # the protein is 0 (or close).
-    def pi(self):
-        """Calculate and return the isoelectric point as float."""
-        # Bracket between pH1 and pH2
-        pH = 7
-        charge = self._chargeR(pH)
-        if charge > 0.0:
-            pH1 = pH
-            charge1 = charge
-            while charge1 > 0.0:
-                pH = pH1 + 1.0
-                charge = self._chargeR(pH)
-                if charge > 0.0:
-                    pH1 = pH
-                    charge1 = charge
-                else:
-                    pH2 = pH
-                    charge2 = charge
-                    break
-        else:
-            pH2 = pH
-            charge2 = charge
-            while charge2 < 0.0:
-                pH = pH2 - 1.0
-                charge = self._chargeR(pH)
-                if charge < 0.0:
-                    pH2 = pH
-                    charge2 = charge
-                else:
-                    pH1 = pH
-                    charge1 = charge
-                    break
-
-        # Bisection
-        while pH2 - pH1 > 0.0001 and charge != 0.0:
-            pH = (pH1 + pH2) / 2.0
-            charge = self._chargeR(pH)
+    def pi(self, pH=7.775, min_=4.05, max_=12):
```

Comment:
How about using ``minimum`` and ``maximum`` to avoid clashing with the built in ``min`` and ``max``?